packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -223,6 +261,11 @@ export class SOQLEditorInstance {
   protected async retrieveSObjects(): Promise<void> {
     return withSFConnection(async conn => {
       conn.describeGlobal$((err, describeGlobalResult) => {
+        if (err) {
+          showAndTrackError('retrieve_sobjects', err.toString()).catch(
```

Comment:
Like Luis pointed out previously, these error messages are not user friendly. Also, I wonder how showing these errors as a toast notification would help customers. I even feel we shouldn't be tracking these errors in telemetry unless we can say how we intend to use this data to make the product better. We already know that a bad/expired connection would cause exceptions/errors to be thrown. I feel it should be enough to log these errors in console for debugging purposes.

Reply:
@lcampos - 
1. What is the standard here so we can be consistent with other extensions?
     a. What error should be in a toast vs in the channel vs just logged in console ( i'm not sure where that goes actually )?
2. Should we track these errors in telemetry at all?

Reply:
@dehru here's what we normally do:

- Use info message for command executions (e.g. create a scratch org)
- Use Output channel for more detailed information (e.g. detailed information when creating a scratch org, detailed information about an error)
- Use error message if a command execution ran into an error (e.g. scratch org failed to get created)
- Use error message if a fundamental setup config is incorrect/missing (e.g. JAVA_HOME is empty and the Apex language server can't run without it)
- Track important events in telemetry (e.g. extension activation, command execution details like elapsed time, etc)
- Do not log any data that could cause a GDPR violation
- Track information that will allow you to see if a feature is being used, if customers are successful with it or constantly running into bugs/errors